### PR TITLE
Fix undefined highlight fallback in vim textprop backend

### PR DESCRIPTION
**Pull request by sam-mccall**
None/NONE are not actually builtin highlight groups, but rather reserved
names that act like them sometimes.
So prop_type_add('foo', {'highlight': 'None'}) is an error. Omitting the 'highlight'
key is fine, though.

Tested with coc.nvim + coc-clangd + clangd 11 + vim 8.1.2269

### CODE REVIEW

**Comment by sam-mccall** [comment]
FWIW, I was seeing errors with coc-clangd immediately after typing "int" on a new line, initially `E970: Unknown highlight group name: 'None'` and afterwards `E716: Key not present in Dictionary: highlight`.

But the error from `prop_type_add('foo', {'highlight': 'None'})` in `vim --clean` is enough to show why the fix is correct I think.

File: autoload/lsp_cxx_hl/textprop.vim [hunk 1]
```diff
@@ -3,12 +3,15 @@
...
-                \ 'highlight': a:resolved_hl_group,
+    let l:opts = {
                 \ 'start_incl': 1,
                 \ 'end_incl': 0,
                 \ 'priority': g:lsp_cxx_hl_syntax_priority
-                \ })
+                \ }
+    if !empty(a:resolved_hl_group)
+       let l:opts['highlight'] = a:resolved_hl_group
+     endif
```
Comment (jackguo380) [inline autoload/lsp_cxx_hl/textprop.vim:13]: extra space here

**Comment by sam-mccall** [comment]
Thanks! Really enjoying the plugin now BTW!

**Comment by jackguo380** [comment]
Awesome, glad to hear its working well for you!

**Pull request merged**